rtf

```diff
@@ -130,6 +133,53 @@ class Vagrant(Host):
     """ Destroy this host """
     os.system("vagrant destroy -f " + self.hostid)
 
+  def export(self, directory):
+    """ Export this VM using Virtualbox commads """
+    fullname = os.path.basename(os.getcwd()).replace('.', '') + '_' + self.hostid
+    # List VMs and do something on the one we are working on
+    for line in os.popen("VBoxManage list vms"):
+      m = re.match(r'"' + fullname + r'[0-9_]+" \{(.*)\}', line)
+      if m:
+        uuid = m.group(1)
+        running = False
+        print("Exporting " + fullname + " / " + uuid + " to directory " + directory)
+        # If the VM is running, pause it (and save its state) and rerun it later
+        for running_line in os.popen("VBoxManage list runningvms"):
+          if re.search(uuid, running_line):
+            running = True
+        if running:
+          os.system("VBoxManage controlvm " + uuid + " savestate")
+        else:
+          print("Currently only running VMs can be exported.")
+          print("If you wish to export a stopped VM, please open ticket to ask for the feature.")
```

Comment:
The function continues if the vm is running, is it ok ? (the print statement mkes me think it should not!!)